### PR TITLE
Project Metrics: Add tests for API modules

### PR DESCRIPTION
**Pull request by rcebulko**
These are not beautiful, but they make it easy to compare to the API docs and confirm that the right endpoints are being called in the right way.

### CODE REVIEW

File: project-metrics/metrics_service/apis/codecov_test.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  def setUp(self):
+    super(TestCodecovApi, self).setUp()
+
+    mock.patch.object(
+        env, 'get', side_effect={
+            'GITHUB_REPO': '__repo__',
+        }.get).start()
+    self.mock_request = mock.patch.object(
+        agithub_base.Client, 'request', autospec=True).start()
```
Comment (danielrozenberg) [inline project-metrics/metrics_service/apis/codecov_test.py]: You know what I think about mocking an implementation detail of a 3rd party library but alright ;)

File: project-metrics/metrics_service/apis/codecov_test.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    self.addCleanup(mock.patch.stopall)
+
+    self.codecov_api = codecov.CodecovApi()
+
+  def testGetAbsoluteCoverage(self):
+    self.mock_request.return_value = (200, {
+        'commit': {
+            'totals': {
+                'c': '13.37'
```
Comment (danielrozenberg) [inline project-metrics/metrics_service/apis/codecov_test.py]: nit: comma if you use this format. Without a comma this should go in one line: `'totals': {'c': '13.37'},` ...

File: project-metrics/metrics_service/apis/travis_test.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    self.mock_request.assert_called_once_with(
+        travis_api.client, 'POST', '/auth/github?github_token=__github_token__',
+        mock.ANY, mock.ANY)
+
+  def testGetTravisTokenError(self):
+    self.mock_request.return_value = (403, 'Unauthorized.')
+
+    with self.assertRaisesRegex(
+        travis.TravisApiError,
+        r'Travis API Exception \(HTTP 403\): Travis Auth API request failed with response: Unauthorized\.'
```
Comment (danielrozenberg) [inline project-metrics/metrics_service/apis/travis_test.py]: You don't need the whole text if you want it to fit in one line ;)

File: project-metrics/metrics_service/apis/codecov_test.py (hunk 1 above)
Comment (rcebulko) [reply project-metrics/metrics_service/apis/codecov_test.py]: Yep yep :)

File: project-metrics/metrics_service/apis/codecov_test.py (hunk 2 above)
Comment (rcebulko) [reply project-metrics/metrics_service/apis/codecov_test.py]: Updated; I use the autoformatter generally, but sometimes I does stuff like this and I have to disable it :unamused:

**Pull request merged**